From b0a98a8c63707706b98bdc66e4d7c460d9efe59e Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 24 May 2022 19:52:58 +0300
Subject: [PATCH] [SW-0] gaudi3: configure lane clock mode in sris mode

In SRIS mode, the pipe lane refclk mode needs to be set to 0 in order
to configure link successfully. This patch adds required configuration
for SRIS mode.

Change-Id: Ieaf060529af2f5126b7228b10e437a3e985497a4
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 1f384d1..2bb4b86 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -262,6 +262,11 @@
 			mmD0_PCIE_PHY_PHYX_MPLLA_SSC_EN_3, 0x1);
 }
 
+void pcie_phy_set_lane_refclk_mode(void)
+{
+	regmap_writel(0x0, mmD0_PCIE_PHY_PIPE_LANE_CMN_REFCLK_MODE);
+}
+
 uint8_t get_sris_mode_from_strap(void)
 {
 	uint32_t boot_strap;
diff --git a/common/pcie.c b/common/pcie.c
index 2c36b25..92c55a4 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -73,6 +73,7 @@
 __weak void pcie_phy_ssc_en_set(void) {}
 __weak void pcie_set_pcie_device_vendor_id(void) {}
 __weak void pcie_config_boot_termination(void) {}
+__weak void pcie_phy_set_lane_refclk_mode(void) {}
 
 void pcie_config(void)
 {
@@ -91,6 +92,7 @@
 		pcie_phy_para_sel_set();
 
 		if (get_sris_mode_from_strap()) {
+			pcie_phy_set_lane_refclk_mode();
 			regmap_writel(1, mmPIPE_RX_SRIS_MODE_EN);
 			regmap_writel(1, mmAPP_SRIS_MODE);
 		}
